src/controller/c/supervisor.c

```diff
@@ -89,21 +89,26 @@ static WbFieldRequest *field_requests_garbage_list = NULL;
 static WbFieldRequest *sent_field_get_request = NULL;
 static bool is_field_immediate_message = false;
 
+typedef struct WbNodeWbContactPointListStructPrivate {
+  int n;
+  WbContactPoint *points;
+  double timestamp;  // TODO: Delete with `wb_supervisor_node_get_contact_point`
+  int sampling_period;
+  double last_update;
+} WbNodeWbContactPointListStruct;
+
 typedef struct WbNodeStructPrivate {
   int id;
   WbNodeType type;
   char *model_name;
   char *def_name;
   char *content;
   int parent_id;
-  double *position;        // double[3]
-  double *orientation;     // double[9]
-  double *center_of_mass;  // double[3]
-  int number_of_contact_points;
-  double *contact_points;           // double[3 * number_of_contact_points]
-  int *node_id_per_contact_points;  // int[number_of_contact_points]
-  double contact_points_time_stamp;
-  bool contact_points_include_descendants;
+  double *position;                                  // double[3]
+  double *orientation;                               // double[9]
+  double *center_of_mass;                            // double[3]
+  WbNodeWbContactPointListStruct contact_points[2];  // 0 -> without descendants, 1 -> with descendants
+  bool contact_points_include_descendants;           // TODO: Delete with `wb_supervisor_node_get_contact_point`
```

Comment:
Can't we address this TODO?

Reply:
Those are just guidelines when deleting the `wb_supervisor_node_get_contact_point` function